command/operator_migrate.go

```diff
@@ -359,11 +366,17 @@ func dfsScan(ctx context.Context, source physical.Backend, cb func(ctx context.C
 				}
 			}
 		} else {
-			err := cb(ctx, key)
+			errorCh := make(chan error)
+			wg.Add(1)
+			go func() {
+				errorCh <- cb(ctx, key)
+				wg.Done()
+			}()
+
+			err := <- errorCh
```

Comment:
This still keeps everything serialized, blocking until the callback returns. I don't think concurrent invocations of `cb` will be running as result